### ISSUE TITLE
Openvstorage id should be cached

### ISSUE THREAD

**Issue by JeffreyDevloo** [openvstorage/framework#2146]
As seen in scrubber logs:
```
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]: 2018-08-10 19:02:16 61200 -0400 - NY2SRV0003 - 29489/140349567391488 - lib/generic tasks scrub - 11208 - ERROR - Scrubber aea34e77-f22f-4837-a686-e085d55d49aa - vPool data-g-03 - StorageRouter NY1SRV1000 - Stack 0 - Scrubbing thread 4 - Scrubbing failed
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]: Traceback (most recent call last):
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/opt/OpenvStorage/ovs/lib/helpers/generic/scrubber.py", line 648, in _execute_scrub
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     self.stack_work_handler.unregister_vdisk(vdisk_guid)
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/opt/OpenvStorage/ovs/lib/helpers/generic/scrubber.py", line 369, in unregister_vdisk
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     max_retries=retries)
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/opt/OpenvStorage/ovs/lib/helpers/generic/scrubber.py", line 129, in _safely_store
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     self._persistent.apply_callback_transaction(build_transaction, max_retries=max_retries, retry_wait_function=retry_callback)
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/usr/lib/python2.7/dist-packages/ovs_extensions/storage/persistent/pyrakoonstore.py", line 44, in new_function
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     return f(self, *args, **kw)
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/usr/lib/python2.7/dist-packages/ovs_extensions/storage/persistent/pyrakoonstore.py", line 204, in apply_callback_transaction
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     return self._client.apply_callback_transaction(transaction_callback, max_retries, retry_wait_function)
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/usr/lib/python2.7/dist-packages/ovs_extensions/db/arakoon/pyrakoon/client.py", line 49, in new_function
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     return f(self, *args, **kw)
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/usr/lib/python2.7/dist-packages/ovs_extensions/db/arakoon/pyrakoon/client.py", line 474, in apply_callback_transaction
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     return apply_callback_transaction(self)
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/usr/lib/python2.7/dist-packages/ovs_extensions/db/arakoon/pyrakoon/client.py", line 86, in wrapped
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     result = f(self, *args, **kwargs)
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/usr/lib/python2.7/dist-packages/ovs_extensions/db/arakoon/pyrakoon/client.py", line 460, in apply_callback_transaction
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     transaction = transaction_callback()
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/opt/OpenvStorage/ovs/lib/helpers/generic/scrubber.py", line 118, in build_transaction
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     value, expected_value = get_value_and_expected_value()
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/opt/OpenvStorage/ovs/lib/helpers/generic/scrubber.py", line 360, in _get_value_and_expected_value
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     relevant_work_items, fetched_work_items = self._unregister_vdisk(vdisk_guid)
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/opt/OpenvStorage/ovs/lib/helpers/generic/scrubber.py", line 334, in _unregister_vdisk
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     relevant_work_items, fetched_work_items = self._get_pending_scrub_work()
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/opt/OpenvStorage/ovs/lib/helpers/generic/scrubber.py", line 292, in _get_pending_scrub_work
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     return self._get_relevant_items(relevant_values=self.worker_contexts.values(), relevant_keys=self.worker_context.keys())
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/opt/OpenvStorage/ovs/lib/helpers/generic/scrubber.py", line 94, in worker_context
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     return self.worker_contexts[System.get_my_storagerouter()]
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/opt/OpenvStorage/ovs/extensions/generic/system.py", line 83, in get_my_storagerouter
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     storagerouter = StorageRouterList.get_by_machine_id(cls.get_my_machine_id())
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:   File "/opt/OpenvStorage/ovs/extensions/generic/system.py", line 54, in get_my_machine_id
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]:     with open(cls.OVS_ID_FILE, 'r') as the_file:
Aug 10 19:02:16 NY2SRV0003 ovs-workers[29462]: IOError: [Errno 24] Too many open files: '/etc/openvstorage_id'

```

**Comment by simonOpenV** [comment]
Fixed by:
https://github.com/openvstorage/framework/pull/2153

**Comment by JeffreyDevloo** [comment]
Fixed by #2153
Packaged in openvstorage-core_2.9.20-1_amd64.deb
Released in https://github.com/openvstorage/framework/releases/tag/2.9.20
